packages/core/src/fhir-to-cda/cda-templates/commons.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  Organization,
+  Address,
+  ContactPoint,
+  CodeableConcept,
+  Identifier,
+} from "@medplum/fhirtypes";
+import {
+  Entry,
+  EntryObject,
+  CDACodeCE,
+  CDACodeCV,
+  CDAInstanceIdentifier,
+  CDAAddress,
+  CDAOrganization,
+  CDATelecom,
+} from "../cda-types/shared-types";
+import {
+  rootAttribute,
+  extensionAttribute,
+  assigningAuthorityNameAttribute,
+  valueAttribute,
+  useAttribute,
+  nullFlavorAttribute,
+} from "./constants";
+
+export function withoutNullFlavorObject(value: string | undefined, key: string): EntryObject {
+  if (value === undefined) {
+    return {};
+  } else {
+    return { [key]: value };
+  }
+}
+
+export function withoutNullFlavorString(value: string | undefined): Entry {
+  if (value === undefined) {
+    return {};
+  } else {
+    return value;
+  }
+}
```

Comment:
I can do that - this involves quite a few of downstream changes...

But are you also suggesting to get replace the Entry type definition to not be this? 
```{ [key: string]: string } | string```

Reply:
Yeah, I talked about this [here](https://github.com/metriport/metriport/pull/1757#discussion_r1571082207). If we say that all properties can either be `string` or `Record<string, string>`, then when we read a property and need to work with it we'll need to check and build for both situations.

`Address.city` is `string`, right? Unless it can actually contain anything, I'd build types accordingly.

For FHIR, we have types generated off of the spec. Wondering if we could leverage the same for CDA?

I tried cxsd, but [it failed](https://github.com/charto/cxsd/issues/33).

If there's no other way, fine, just wanted to make sure we're not losing the opportunity to make this easier to maintain.

Reply:
So for this, I changed the types of fields that should just be `string | undefined` to match that. But there are other fields that do show up as either `string or Record<string, string>`, in CDAs. So I kept that for now.

Reply:
Also had a discussion about this on a call, and decided to keep the rest of them as is, since we're only just sending these more complicated object/strings to the xml generator and not trying to work with those objects in mem.